.circleci/config.yml

```diff
@@ -437,35 +450,54 @@ jobs:
             --form "file=@${fileName}" \
 
   instrumentation_tests_device:
-    macos:
-      xcode: "12.3.0"
-    resource_class: large
+    resource_class: xlarge
+    working_directory: ~/project
+    docker:
+      - image: circleci/android:api-28
     steps:
+      - skip-for-external-pull-requests
+      - setup-google-cloud
       - checkout
-      - setup-android-macos
       - restore-gradle-cache
-      - restore-android-build-cache-macos
-      - run-gradle-cmd:
-          desc: Build Apks for instrumentation test
-          cmd: >
-            :Corona-Warn-App:assembleDeviceDebug
-            :Corona-Warn-App:assembleDeviceDebugAndroidTest
-      - save-gradle-cache
-      - save-android-build-cache-macos
-      - run-emulator-for-api:
-          apilevel: 29
-      - run-gradle-cmd:
-          desc: Run instrumentation tests on device
-          cmd: >
-            :Corona-Warn-App:connectedDeviceDebugAndroidTest
-            --info --continue
-            -Pandroid.testInstrumentationRunnerArguments.notAnnotation=testhelpers.Screenshot
-            -Pandroid.testInstrumentationRunnerArguments.clearPackageData=true
-      - kill-all-emulators
+      - restore-android-build-cache
+      - run:
+          name: "Build app APK "
+          command: ./gradlew -PdisablePreDex :Corona-Warn-App:assembleDeviceForTestersDebug
+      - run:
```

Comment:
Could we use our `run-gradle-cmd` which automatically adds a few optimization flags, or did you try that and there were downsides?

Reply:
will try it